internal/executor/instance/persistentworker/isolation/tart/cmd.go

```diff
@@ -25,7 +25,7 @@ type loggerAsWriter struct {
 
 func (l loggerAsWriter) Write(p []byte) (n int, err error) {
 	if l.delegate != nil {
-		l.delegate.Logf(l.level, string(p))
+		l.delegate.Logf(l.level, "%s", strings.TrimSpace(string(p)))
```

Comment:
> Is there a need for the string interpolation?

Do you mean `string(p)`? Without it, we'l print a `[]byte` as raw bytes, which is not what we want here.

>What was the reason for the weird log lines?

`Logf()`'s second argument is a format string, but we were passing a value.

If that value contains formatting directives, which normally start with `%`, this will cause the function to look for corresponding values to format, which we don't provide, and this results in `(MISSING)`.

Reply:
Got it. Thank you! Do we need to trim things though? What if spaces are legit in the chunk?

Reply:
I don't think we have a case when multiple newlines are actually used to convey any information in Tart.

And `Logf` already adds a newline at the end.

Without trimming the output looks like this:

```
% ~/cirrus run -o simple -v
gRPC server is listening at http://127.0.0.1:59606 (http://host.docker.internal:59606 inside of a container)
running task main (0)
Started 'main' task
Started 'pull virtual machine'
Pulling virtual machine ghcr.io/cirruslabs/macos-monterey-base:latest...
pulling ghcr.io/cirruslabs/macos-monterey-base:latest...

pulling manifest...

pulling disk (18.3 GB compressed)...

0%

3%
```

I'm not sure if that's what we want?

With trimming:

```
gRPC server is listening at http://127.0.0.1:59608 (http://host.docker.internal:59608 inside of a container)
running task main (0)
Started 'main' task
Started 'pull virtual machine'
Pulling virtual machine ghcr.io/cirruslabs/macos-monterey-base:latest...
pulling ghcr.io/cirruslabs/macos-monterey-base:latest...
pulling manifest...
pulling disk (18.3 GB compressed)...
3%
```